h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -632,36 +632,41 @@ public static MVStore open(String fileName) {
      * @return the map
      */
     public <M extends MVMap<K, V>, K, V> M openMap(String name, MVMap.MapBuilder<M, K, V> builder) {
-        int id = getMapId(name);
-        if (id >= 0) {
-            @SuppressWarnings("unchecked")
-            M map = (M) getMap(id);
-            if(map == null) {
-                map = openMap(id, builder);
-            }
-            assert builder.getKeyType() == null || map.getKeyType().getClass().equals(builder.getKeyType().getClass());
-            assert builder.getValueType() == null
-                    || map.getValueType().getClass().equals(builder.getValueType().getClass());
-            return map;
-        } else {
-            HashMap<String, Object> c = new HashMap<>();
-            id = lastMapId.incrementAndGet();
-            assert getMap(id) == null;
-            c.put("id", id);
-            c.put("createVersion", currentVersion);
-            M map = builder.create(this, c);
-            String x = Integer.toHexString(id);
-            meta.put(MVMap.getMapKey(id), map.asString(name));
-            meta.put(DataUtils.META_NAME + name, x);
-            long lastStoredVersion = currentVersion - 1;
-            map.setRootPos(0, lastStoredVersion);
-            markMetaChanged();
-            @SuppressWarnings("unchecked")
-            M existingMap = (M) maps.putIfAbsent(id, map);
-            if (existingMap != null) {
-                map = existingMap;
+        storeLock.lock();
```

Comment:
This will block even if map exists already, which might be a performance penalty, given a wide scope of that lock.